src/pages/studyView/UserSelections.tsx

```diff
@@ -0,0 +1,94 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { observer } from "mobx-react";
+import { computed } from 'mobx';
+import styles from "./styles.module.scss";
+import { StudyViewFilter } from 'shared/api/generated/CBioPortalAPIInternal';
+import { ChartMeta } from 'pages/studyView/StudyViewPageStore';
+import { isFiltered } from 'pages/studyView/StudyViewUtils';
+
+export interface IUserSelectionsProps {
+    filter: StudyViewFilter;
+    attributesMetaSet: { [id: string]: ChartMeta };
+    updateClinicalDataEqualityFilter: (chartMeta: ChartMeta, value: string[]) => void;
+    clearGeneFilter: () => void;
+    clearCNAGeneFilter: () => void;
+    clearCustomCasesFilter: () => void;
+    clearAllFilters: () => void
+}
+
+
+@observer
+export default class UserSelections extends React.Component<IUserSelectionsProps, {}> {
+
+    constructor(props: IUserSelectionsProps) {
+        super(props);
+    }
+
+    @computed get showFilters() {
+        return isFiltered(this.props.filter)
+    }
+
+    render() {
+        if (this.showFilters) {
+            return (
+                <div className={styles.studyViewUserSelections}>
+                    <span>Your selection:</span>
+                    <div>
+                        {
+                            _.map((this.props.filter.clinicalDataEqualityFilters || []), clinicalDataEqualityFilter => {
```

Comment:
this code is too deeply nested.  i would remove it to computed getter or even some nested components.  look at stateless components for example